components/support/nimbus-fml/src/backends/kotlin/templates/FeatureManifestTemplate.kt

```diff
@@ -95,6 +96,21 @@ object {{ nimbus_object }} : FeatureManifestInterface<{{ nimbus_object }}.Featur
         {%- endfor %}
     }
 
+    /**
+     * Get a list of feature ids where the feature allows co-enrollment.
+     */
+    public override fun getCoenrolledFeatureIds(): List<String>? {
+        {%- let features = self.iter_feature_defs() %}
+        {%- if features.is_empty() %}
+        return emptyList()
+        {%- let coenrolledFeatures = ArrayList<String> %}
+        {%- for f in features %}
+        {%- if f.allow_coenrollment() %}
+        coenrolledFeatures += features.{{- f.name() -}}
+        {%- endfor %}
+        return coenrolledFeatures
+    }
```

Comment:
A few comments here:
- I'd prefer to standardize on one of `conenrolling` or `coenrolled`; #5684 at the moment uses `coenrolling`; let's pick one and stick to it.
- The return type doesn't need to be optional.
- I think you're doing the hard work in `get_coenrolling_feature_ids`; no need to duplicate it in the templates.

```suggestion
    override fun getCoenrollingFeatureIds() = listOf(
            {%- for f in self.fm.get_coenrolling_feature_ids() %}{{ f|quoted }}
            {%- if !loop.last %}, {% endif %}
            {%- endfor %})
```

Reply:
what is `{{ f|quoted }}` doing here?

Reply:
It's a pipe similar to what you might see in other templating frameworks — Angular/Helm both have them, as examples. 

`quoted` is a function that's declared in the `gen_structs/common` file, and the pipe passes the left-hand value into the function

https://github.com/mozilla/application-services/blob/c2823f590b26c259becd103e76050bbff1d272f2/components/support/nimbus-fml/src/backends/kotlin/gen_structs/common.rs#L22-L32

Reply:
I'm realizing now you may have been asking _why_ it's there and not what it is doing, so sorry

Reply:
no, you were right! thank you so much for the explanation :D